pkg/controller/common.go

```diff
@@ -597,7 +589,7 @@ func (c *SharedState) isVolumeVisited(azVolumeName string) bool {
 	return visited
 }
 
-func getRankedNodesForReplicaAttachments(ctx context.Context, azr azReconciler, volumes, pods []string) ([]string, error) {
+func getRankedNodesForReplicaAttachments(ctx context.Context, azr azReconciler, volumes []string, podsObjs ...v1.Pod) ([]string, error) {
```

Comment:
Why take Pods as a variable number of arguments vs. a slice (i.e. `pods []v1.Pod`)?

Reply:
Changed